apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx

```diff
@@ -0,0 +1,396 @@
+import React from 'react'
+import { useCallback, useMemo, useRef } from 'react'
+
+import ICON_ARROW from '@cowprotocol/assets/cow-swap/arrow.svg'
+import ICON_CURVE from '@cowprotocol/assets/cow-swap/icon-curve.svg'
+import ICON_PANCAKESWAP from '@cowprotocol/assets/cow-swap/icon-pancakeswap.svg'
+import ICON_SUSHISWAP from '@cowprotocol/assets/cow-swap/icon-sushi.svg'
+import ICON_UNISWAP from '@cowprotocol/assets/cow-swap/icon-uni.svg'
+import ICON_STAR from '@cowprotocol/assets/cow-swap/star-shine.svg'
+import { ProductLogo, ProductVariant, UI } from '@cowprotocol/ui'
+
+import SVG from 'react-inlinesvg'
+import { Textfit } from 'react-textfit'
+
+import { upToSmall, useMediaQuery } from 'legacy/hooks/useMediaQuery'
+import { useIsDarkMode } from 'legacy/state/user/hooks'
+
+import { ArrowBackground } from './arrowBackground'
+import { LpToken, StateKey, dummyData, lpTokenConfig } from './dummyData'
+import * as styledEl from './styled'
+
+import { BannerLocation, DEMO_DROPDOWN_OPTIONS } from './index'
+import { TokenLogo } from '../../../../../../libs/tokens/src/pure/TokenLogo'
+import { USDC, WBTC } from '@cowprotocol/common-const'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { DummyDataType, TwoLpScenario, InferiorYieldScenario } from './dummyData'
+
+const lpTokenIcons: Record<LpToken, string> = {
+  [LpToken.UniswapV2]: ICON_UNISWAP,
+  [LpToken.Sushiswap]: ICON_SUSHISWAP,
+  [LpToken.PancakeSwap]: ICON_PANCAKESWAP,
+  [LpToken.Curve]: ICON_CURVE,
+}
+
+interface CoWAmmBannerContentProps {
+  id: string
+  title: string
+  ctaText: string
+  location: BannerLocation
+  isDemo: boolean
+  selectedState: StateKey
+  setSelectedState: (state: StateKey) => void
+  dummyData: typeof dummyData
+  lpTokenConfig: typeof lpTokenConfig
+  onCtaClick: () => void
+  onClose: () => void
+}
+
+function isTwoLpScenario(scenario: DummyDataType[keyof DummyDataType]): scenario is TwoLpScenario {
+  return 'uniV2Apr' in scenario && 'sushiApr' in scenario
+}
+
+function isInferiorYieldScenario(scenario: DummyDataType[keyof DummyDataType]): scenario is InferiorYieldScenario {
+  return 'poolsCount' in scenario
+}
+
+const renderTextfit = (
+  content: React.ReactNode,
+  mode: 'single' | 'multi',
+  minFontSize: number,
+  maxFontSize: number,
+  key: string,
+) => (
+  <Textfit key={key} mode={mode} forceSingleModeWidth={false} min={minFontSize} max={maxFontSize}>
+    {content}
+  </Textfit>
+)
+
+export function CoWAmmBannerContent({
+  id,
+  title,
+  ctaText,
+  location,
+  isDemo,
+  selectedState,
+  setSelectedState,
+  dummyData,
+  lpTokenConfig,
+  onCtaClick,
+  onClose,
+}: CoWAmmBannerContentProps) {
+  const isMobile = useMediaQuery(upToSmall)
+  const isDarkMode = useIsDarkMode()
```

Comment:
Ideally these should come as props, given this is a pure component

Reply:
I will refactor that part along with connecting it to the logic